cart/domain/cart/cartServicePorts.go

```diff
@@ -98,10 +98,12 @@ var (
 
 //CreateDeliveryInfoUpdateCommand - factory to get the update command based on the given deliveryInfos (which might come from cart)
 func CreateDeliveryInfoUpdateCommand(info DeliveryInfo) DeliveryInfoUpdateCommand {
-	return DeliveryInfoUpdateCommand{
+	uc := DeliveryInfoUpdateCommand{
 		DeliveryInfo: info,
 		additional:   info.AdditionalDeliveryInfos,
 	}
+	uc.SetAdditional(info.AdditionalDeliveryInfos)
+	return uc
 }
```

Comment:
I think we can remove line 103 as we always overwrite the value of additional in SetAdditional.

From a first glance it is a bit confusing why that value is assigned twice. 
First when initialising the struct and then again using the function. 

When checking out init I understood that we do not want that value to be nil.

Reply:
I'll updated the code, it should now make more sense.